controllers/core/openstackcontrolplane_controller.go

```diff
@@ -170,10 +172,81 @@ func (r *OpenStackControlPlaneReconciler) Reconcile(ctx context.Context, req ctr
 		return ctrl.Result{}, nil
 	}
 
-	return r.reconcileNormal(ctx, instance, helper)
+	Log.Info("Looking up the current OpenStackVersion")
+	ctrlResult, err, version := openstack.ReconcileVersion(ctx, instance, helper)
+	if err != nil {
+		return ctrl.Result{}, err
+	} else if (ctrlResult != ctrl.Result{}) {
+		return ctrlResult, nil
+	}
+
+	// wait until the version is initialized so we have images on the version.Status
+	if version.Status.Conditions.IsFalse(corev1beta1.OpenStackVersionInitialized) {
+		return ctrlResult, nil
+	}
+
+	if instance.Status.DeployedVersion == nil || version.Spec.TargetVersion == *instance.Status.DeployedVersion {
+		// green field deployment or no minor update in progress
+		ctrlResult, err := r.reconcileNormal(ctx, instance, version, helper)
+		if err != nil {
+			Log.Info("Error reconciling normal", "error", err)
+			return ctrl.Result{}, err
+		} else if (ctrlResult != ctrl.Result{}) {
+			Log.Info("Reconciling normal")
+			return ctrlResult, nil
+		}
+		// this will allow reconcileNormal to proceed in subsequent reconciles
+		instance.Status.DeployedVersion = &version.Spec.TargetVersion
+		return ctrl.Result{}, nil
+	} else {
+		if version.Status.Conditions.IsFalse(corev1beta1.OpenStackVersionMinorUpdateOVNControlplane) {
+			Log.Info("Minor update OVN on the ControlPlane")
+			ctrlResult, err := r.reconcileOVNControllers(ctx, instance, version, helper)
+			if err != nil {
+				return ctrl.Result{}, err
+			} else if (ctrlResult != ctrl.Result{}) {
+				return ctrlResult, nil
+			}
+			instance.Status.DeployedOVNVersion = &version.Spec.TargetVersion
+			return ctrl.Result{}, nil
+		} else if version.Status.Conditions.IsFalse(corev1beta1.OpenStackVersionMinorUpdateControlplane) {
+			Log.Info("Minor update on the ControlPlane")
+			ctrlResult, err := r.reconcileNormal(ctx, instance, version, helper)
+			if err != nil {
+				return ctrl.Result{}, err
+			} else if (ctrlResult != ctrl.Result{}) {
+				return ctrlResult, nil
+			}
+			// this will allow reconcileNormal to proceed in subsequent reconciles
+			instance.Status.DeployedVersion = &version.Spec.TargetVersion
+			return ctrl.Result{}, nil
+		} else {
+			Log.Info("Skipping reconcile. Waiting on minor update to proceed.")
+			return ctrl.Result{}, nil
+		}
+	}
+
+}
+
+func (r *OpenStackControlPlaneReconciler) reconcileOVNControllers(ctx context.Context, instance *corev1beta1.OpenStackControlPlane, version *corev1beta1.OpenStackVersion, helper *helper.Helper) (ctrl.Result, error) {
+
+	OVNControllerReady, err := openstack.ReconcileOVNController(ctx, instance, version, helper)
+	if err != nil {
+		return ctrl.Result{}, err
+	} else if !OVNControllerReady {
+		instance.Status.Conditions.Set(condition.FalseCondition(
+			corev1beta1.OpenStackControlPlaneOVNReadyCondition,
+			condition.RequestedReason,
+			condition.SeverityInfo,
+			corev1beta1.OpenStackControlPlaneOVNReadyRunningMessage))
```

Comment:
I would mirror the message of the Ready conditon form OVN to propagate the reason. https://github.com/openstack-k8s-operators/lib-common/blob/fec16b14c33be8622459fa2a652e5553d4912e68/modules/common/condition/funcs.go#L321